docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
+
+*Notes* is an optional section that provides additional information about the code, possibly including a discussion of the algorithm or known limitations of the code. The notes should be prefixed by a ``@note`` or ``@warning`` command.
+
+For internal consistency, always use ``@note`` and not the synonymous ``@remark`` or ``@remarks``.
+
+This section may include mathematical equations, written in `LaTeX <http://www.latex-project.org/>`_ format:
+
+.. code-block:: cpp
+
+   /**
+    * The FFT is a fast implementation of the discrete Fourier transform:
+    * @f[ X(e^{j\omega } ) = x(n)e^{ - j\omega n} @f]
+    */
+
+LaTeX environments can also be used:
+
+.. code-block:: cpp
+
+   /**
+    * The discrete-time Fourier time-convolution property states that
+    * @f{eqnarray*}
+    * x(n) * y(n) \Leftrightarrow X(e^{j\omega } )Y(e^{j\omega } )\\
+    * another equation here
+    * @f}
+    */
+
+Math can also be used inline:
+
+.. code-block:: cpp
+
+   /**
+    * The value of @f$\omega@f$ is larger than 5.
+    */
+
+Note that LaTeX is not particularly easy to read, so use equations sparingly. Doxygen recovers poorly from typos in formulas; you may need to manually delete ``docs/html/formula.repository`` if it contains a bad formula.
+
+Images are allowed, but should not be central to the explanation; users viewing the documentation as text must be able to comprehend its meaning without resorting to an image viewer.
+These additional illustrations are included using:
+
+.. code-block:: cpp
+
+   /**
+    * @image html filename ["caption"]
+    */
+
+where ``filename`` is a path relative to the project root directory.
+
+.. _cpp-doxygen-references:
+
+References
+----------
+
+References can be included either in the :ref:`'Notes' <cpp-doxygen-notes>` section or in a separate list below them. A reference consists of the ``@cite`` tag, followed by a BibTeX label. Bibfiles must be listed in the ``CITE_BIB_FILES`` configuration tag in ``doc/doxygen.conf.in``.
+
+Note that Web pages should be referenced with regular inline links.
+
+References are meant to augment the documentation, but should not be required to understand it.
+
+.. _cpp-doxygen-examples:
+
+Examples
+--------
+
+'Examples' is an optional section for examples. This section is very strongly encouraged.
+
+Examples should be prefixed by ``@code`` and suffixed by ``@endcode``:
+
+.. code-block:: cpp
+
+   /**
+    * @code
+    * auto cosines = cos(angles);
+    * @endcode
+    *
+    * Comment explaining the second example.
+    *
+    * @code
+    * auto cosines = cos(radians(angles));
+    * @endcode
+    */
+
+File Description Comment for Header Files
+=========================================
+
+``.h`` files should not contain documentation on the file itself (i.e., a documentation block containing a ``@file`` command). Document the class or package instead, where possible.
+
+.. _cpp-doxygen-package-definition:
+
+Documenting/Defining Packages
+==============================
```

Comment:
I'm wondering how much documentation needs to be written in doxygen in this context. I think we can skip the Extended summary for packages since most documentation at this level will be written directly in Sphinx rST — I'm writing up a description of the Science Pipelines documentation architecture in DMTN-030 as we speak after a meeting we had a week and a half ago.